Source/Core/Common/ArmCPUDetect.cpp

```diff
@@ -9,45 +9,205 @@
+  status = RegGetValueA(HKEY_LOCAL_MACHINE, SUBKEY_CORE0, name.c_str(), flags, nullptr,
+                        value->data(), &value_len);
+  if (status != ERROR_SUCCESS)
+  {
+    value->clear();
+    return false;
+  }
+
+  TruncateToCString(value);
+  return true;
+}
+
+// Read cached register values from the registry
+static bool ReadPrivilegedCPReg(u64* value, u32 reg)
+{
+  DWORD value_len = sizeof(*value);
+  // Not sure if the value name is padded or not
+  return RegGetValueA(HKEY_LOCAL_MACHINE, SUBKEY_CORE0, fmt::format("CP {:x}", reg).c_str(),
+                      RRF_RT_REG_QWORD, nullptr, value, &value_len) == ERROR_SUCCESS;
+}
+
+static bool Read_MIDR_EL1(u64* value)
+{
+  return ReadPrivilegedCPReg(value, ARM64_SYSREG(0b11, 0, 0, 0b0000, 0));
+}
+
+static bool Read_ID_AA64ISAR0_EL1(u64* value)
+{
+  return ReadPrivilegedCPReg(value, ARM64_SYSREG(0b11, 0, 0, 0b0110, 0));
+}
+
+static bool Read_ID_AA64MMFR1_EL1(u64* value)
+{
+  return ReadPrivilegedCPReg(value, ARM64_SYSREG(0b11, 0, 0, 0b0111, 1));
+}
+
+#endif
+
+#if defined(__linux__)
+
+static bool ReadDeviceTree(std::string* value, const std::string& name)
+{
+  const std::string path = std::string("/proc/device-tree/") + name;
   std::ifstream file;
-  File::OpenFStream(file, procfile, std::ios_base::in);
+  File::OpenFStream(file, path.c_str(), std::ios_base::in);
+  if (!file)
+    return false;
 
+  file >> *value;
+  return true;
+}
+
+static std::string ReadCpuinfoField(const std::string& field)
+{
+  std::string line;
+  std::ifstream file;
+  File::OpenFStream(file, "/proc/cpuinfo", std::ios_base::in);
   if (!file)
-    return cpu_string;
+    return {};
 
   while (std::getline(file, line))
   {
-    if (line.find(marker) != std::string::npos)
-    {
-      cpu_string = line.substr(marker.length());
-      break;
-    }
+    if (!StringBeginsWith(line, field))
+      continue;
+    auto non_tab = line.find_first_not_of("\t", field.length());
+    if (non_tab == line.npos)
+      continue;
+    if (line[non_tab] != ':')
+      continue;
+    auto value_start = line.find_first_not_of(" ", non_tab + 1);
+    if (value_start == line.npos)
+      continue;
+    return line.substr(value_start);
   }
+  return {};
+}
 
-  return cpu_string;
+static bool Read_MIDR_EL1_Sysfs(u64* value)
+{
+  std::ifstream file;
+  File::OpenFStream(file, "/sys/devices/system/cpu/cpu0/regs/identification/midr_el1",
+                    std::ios_base::in);
+  if (!file)
+    return false;
+
+  file >> std::hex >> *value;
+  return true;
+}
+
+#endif
+
+#if defined(__linux__) || defined(__FreeBSD__)
+
+static u32 ReadHwCap(u32 type)
+{
+#if defined(__linux__)
+  return getauxval(type);
+#elif defined(__FreeBSD__)
+  u_long hwcap = 0;
+  elf_aux_info(type, &hwcap, sizeof(hwcap));
+  return hwcap;
+#endif
+}
+
+// For "Direct" reads, value gets filled via emulation, hence:
+// "there is no guarantee that the value reflects the processor that it is currently executing on"
```

Comment:
Also, you will get different values when running on big or little cores

Worth mentioning, probably not worth dealing with.

Reply:
yea that's what i was hinting at, maybe ill make it more clear :) (the rest of the methods of getting data all use core 0, so at least it will be consistent)